mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -176,8 +176,12 @@ private[classification] trait LogisticRegressionParams extends ProbabilisticClas
     }
   }
 
-  override def validateParams(): Unit = {
+  override protected def validateAndTransformSchema(
```

Comment:
I think it's a bug that ```validateParams ``` was never used. It should validate params interaction before fitting(if necessary), this is why we deprecate ```validateParams ``` and move what it does to ```transformSchema```. We do not have corresponding test cases before, so no test was broken when we deprecated ```validateParams```. I added test cases in this PR.